File: 141-linked-list-cycle/linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        if not head or not head.next or not head.next.next:
            return False

        i, j = head.next, head.next.next

        while i is not j:
            try:
                i = i.next
                j = j.next.next
            except:
                return False
        
        return True


        

File: 141-linked-list-cycle/linked-list-cycle.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
import "fmt"
func hasCycle(head *ListNode) bool {
    if head == nil {
        return false
    }
    i := head
    j := head

    for j != nil{
        i = i.Next
        j = j.Next

        if j == nil{
            return false
        }
        j = j.Next
        if j == i {
            return true
        }
    }
    return false
}